tests/pkg_tinyvcdiff/fakemtd.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2022 Juergen Fitschen
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @ingroup  tests
+ * @brief    Fake MTD device
+ *
+ * @author   Juergen Fitschen <me@jue.yt>
+ */
+#ifndef FAKEMTD_H
+#define FAKEMTD_H
+
+#include "mtd.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Page size
+ */
+#define FAKE_MTD_PAGE_SIZE        128
+
+/**
+ * @brief Pages per sector
+ *
+ * @note FAKE_MTD_PAGE_SIZE * FAKE_MTD_PAGES_PER_SECTOR must be 512 for FatFS
+ */
+#define FAKE_MTD_PAGES_PER_SECTOR 4
+
+/**
+ * @brief Sector count
+ *
+ * @note FatFS requires more than 128 sectors
```

Comment:
Then let's use something else so we have a chance of running this on some targets too.

<details><summary>littlefs2</summary>

```patch
diff --git a/tests/pkg_tinyvcdiff/Makefile b/tests/pkg_tinyvcdiff/Makefile
index 9667e2b9c9..59bb9b1abd 100644
--- a/tests/pkg_tinyvcdiff/Makefile
+++ b/tests/pkg_tinyvcdiff/Makefile
@@ -12,7 +12,7 @@ USEPKG += tinyvcdiff
 USEMODULE += mtd
 
 # VFS backend
-USEPKG += fatfs
+USEPKG += littlefs2
 USEMODULE += fatfs_vfs
 USEMODULE += fatfs_vfs_format
 
diff --git a/tests/pkg_tinyvcdiff/fakemtd.h b/tests/pkg_tinyvcdiff/fakemtd.h
index 1c3a0d8302..e6a20f9e70 100644
--- a/tests/pkg_tinyvcdiff/fakemtd.h
+++ b/tests/pkg_tinyvcdiff/fakemtd.h
@@ -31,17 +31,13 @@ extern "C" {
 
 /**
  * @brief Pages per sector
- *
- * @note FAKE_MTD_PAGE_SIZE * FAKE_MTD_PAGES_PER_SECTOR must be 512 for FatFS
  */
 #define FAKE_MTD_PAGES_PER_SECTOR 4
 
 /**
  * @brief Sector count
- *
- * @note FatFS requires more than 128 sectors
  */
-#define FAKE_MTD_SECTOR_COUNT     256
+#define FAKE_MTD_SECTOR_COUNT     4
 
 /**
  * @brief Storage size in byte
diff --git a/tests/pkg_tinyvcdiff/main.c b/tests/pkg_tinyvcdiff/main.c
index d2161db690..7b5c859fbd 100644
--- a/tests/pkg_tinyvcdiff/main.c
+++ b/tests/pkg_tinyvcdiff/main.c
@@ -26,7 +26,7 @@
 #include "vcdiff_mtd.h"
 #include "vcdiff_vfs.h"
 #include "fakemtd.h"
-#include "fs/fatfs.h"
+#include "fs/littlefs2_fs.h"
 
 static const char *source_bin = "Hello world! I hope you are doing well ...\n";
 static const char *target_bin = "Hello universe! I hope you are doing well ...\n";
@@ -82,10 +82,10 @@ static void test_tinyvcdiff_vfs(void)
     static vcdiff_t vcdiff;
 
     static fake_mtd_t storage = FAKE_MTD_INIT;
-    fatfs_desc_t fs = { .dev = &storage.mtd };
+    littlefs2_desc_t fs = { .dev = &storage.mtd };
     vfs_mount_t mnt = {
         .mount_point = "/mnt",
-        .fs = &fatfs_file_system,
+        .fs = &littlefs2_file_system,
         .private_data = &fs,
     };
     vfs_format(&mnt);
```
</details>

I'm happy to report that this then also works on AVR.

Reply:
Nice! Didn't expected that - the mandatory address cache needs some kB of space in RAM.

Reply:
156b8dd9c125ef150a963216537b063d62753456

And I shared the large context objects between both test cases: 375b241933fd93b8715e58b5e7d1ddb7b8079903